### PR TITLE
Emphasize context/motivation of PRs for contributors

### PR DESCRIPTION
**Pull request by maxhutch**
# Citrine Python PR

## Description 
This adds text to the contributing guide and PR template to
emphasize the responsibility of the PR author to connect to the
context and motivation of the PR.

Also adds a `maintenance` category for the PR type, so PRs like this don't feel out of place.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in setup.py

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -191,6 +191,10 @@ A passing build requires the following:
 * The linter finds no violations of PEP8 style
 * Every line of code is executed by a test (100% coverage)
 
+PR descriptions should describe the motivation and context of the code changes in the PR,
+both for the reviewer and also for future developers.
+If there's a JIRA ticket or Github issue, linking to it is an easy and often effective way to provide that context.
```
Comment (fayvor) [inline CONTRIBUTING.md:196]: I would say:
```suggestion
If there's a JIRA ticket or Github issue, the PR should be linked to the ticket/issue to provide that context.
```

Comment (maxhutch) [reply CONTRIBUTING.md:196]: Personally, I'd rather be as outcome-driven as possible.  If you feel strongly, I'm OK with it, though.

**Comment by kroenlein** [comment]
@fayvor This PR seems to be stalled.  Should it close or ought we take this across the line?

**Review by fayvor — APPROVED** [review]
Looks good!

**Pull request merged**